caas/kubernetes/clientconfig/k8s.go

```diff
@@ -162,3 +161,60 @@ func getKubeConfigPath() string {
 	}
 	return envPath
 }
+
+// getKubeConfig reads kubeconfig file either stdin or config file, then deserializes the contents into Config object.
```

Comment:
Nit: This comment line is too long. Rule of thumb: if you aim to fit it within 80 chars, it'll b readable.

Also comments are mandatory on exported functionality, not necessary on un-exported functions like here. If you find that your comment is redundant and the code is self-explanatory, maybe no need to add it?